admin_app_2/src/utils/auth.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+import { apiCalls } from "@/utils/api";
+import { useRouter, useSearchParams } from "next/navigation";
+import { ReactNode, createContext, useContext, useState } from "react";
+
+type AuthContextType = {
+  token: string | null;
+  user: string | null;
+  accessLevel: "readonly" | "fullaccess";
+  loginError: string | null;
+  login: (username: string, password: string) => void;
+  logout: () => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+type AuthProviderProps = {
+  children: ReactNode;
+};
+
+const AuthProvider = ({ children }: AuthProviderProps) => {
+  const [user, setUser] = useState<string | null>(null);
+  const getInitialToken = () => {
+    if (typeof window !== "undefined") {
+      return localStorage.getItem("token");
+    }
+    return null;
+  };
+
+  const [token, setToken] = useState<string | null>(getInitialToken);
```

Comment:
`useEffect` runs at each render of the component but since it's async, it is not guaranteed to be done when this is being initialized. So when you use `useAuth` it returns `null` since it hasn't updated when the other page (like /content) is being rendered.

I think we leave it as is for now.

Reply:
Hmm interesting - sounds good